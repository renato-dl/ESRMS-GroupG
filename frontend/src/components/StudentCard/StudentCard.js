import React from 'react';
import { Card, Image } from 'semantic-ui-react'
import studentStubImage from '../../assets/images/student-stub.png'
import './StudentCard.scss';

export const StudentCard = (props) => (
  <Card 
    className="student" 
    key={props.id} 
    onClick={props.onClick ? props.onClick : null}
  >
    <Image src={studentStubImage} wrapped ui={false} />
    <Card.Content>
      <Card.Header>{props.firstName} {props.lastName}</Card.Header>
    </Card.Content>
    <Card.Content extra>
      <span>First Class</span>
    </Card.Content>
  </Card>
);
