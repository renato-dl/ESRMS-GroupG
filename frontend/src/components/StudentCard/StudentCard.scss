.ui.card.student {
  width: 200px !important;
  margin: 0 !important;
  margin-right: 25px !important;
  //border:1px solid #008272;


  .image {
    // width: 19200px;
  }
  .studentCardHeader{
    font-size: 22px!important;
  color: #273037!important;
  font-variant: all-small-caps;
  text-align: center;
  }
  div.studentCardExtra{
    float:left;
    font-size:12px;
    color:#009B8B;
  }
}
.ui.card.student:hover{
  
  border:1px solid #008272;
}
