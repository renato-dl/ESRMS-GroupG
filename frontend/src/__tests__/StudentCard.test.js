import React from 'react';
import {render} from '@testing-library/react'
import { StudentCard } from '../components/StudentCard/StudentCard';

describe('Testing StudentCard component', () => {
  
  test('Test if component is rendered', () => {
    const component = render(
      <StudentCard 
        id="1"
        firstName="Name"
        lastName="Surname"
      />,
    );

  });
});
