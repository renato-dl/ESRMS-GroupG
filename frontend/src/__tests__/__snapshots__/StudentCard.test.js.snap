// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing StudentCard component Testing snapshot 1`] = `
<div
  className="ui card student"
  onClick={[Function]}
>
  <div
    className="image"
  >
    <img
      src="student-stub.png"
    />
  </div>
  <div
    className="content"
  >
    <div
      className="header"
    >
      Name
       
      Surname
    </div>
  </div>
  <div
    className="extra content"
  >
    <span>
      First Class
    </span>
  </div>
</div>
`;

exports[`Testing StudentCard component Testing snapshot 2`] = `
<div
  className="ui card student"
  onClick={[Function]}
>
  <div
    className="image"
  >
    <img
      src="student-stub.png"
    />
  </div>
  <div
    className="content"
  >
    <div
      className="header"
    >
      Name
       
      Surname
    </div>
  </div>
  <div
    className="extra content"
  >
    <span>
      First Class
    </span>
  </div>
</div>
`;
